styx-common/src/main/java/com/spotify/styx/util/TimeUtil.java

```diff
@@ -174,6 +175,36 @@ public static Instant nextInstant(Instant instant, Schedule schedule) {
     return instants;
   }
 
+  /**
+   * Gets the instant with the offset to the reference instant.
+   *
+   * @param referenceInstant The start instant from where to go back or forward
+   * @param offset           How far to go back or forward
+   * @param schedule         The schedule of the workflow
+   *
+   * @return The instant with the offset to the reference instant
+   */
+  public static Instant instantWithOffsetTo(Instant referenceInstant, int offset, Schedule schedule) {
+    if (!isAligned(referenceInstant, schedule)) {
+      throw new IllegalArgumentException("unaligned reference instant");
+    }
+
+    final ExecutionTime executionTime = ExecutionTime.forCron(cron(schedule));
+
+    Instant currentInstant = referenceInstant;
+    for (int i = 0; i < Math.abs(offset); i++) {
```

Comment:
Offsetting the instant can be done more efficiently: https://github.com/spotify/styx/pull/591/files#diff-83fdfa5c37381e562bd0705651c031a6R271